src/include/stir/DetectorCoordinateMap.h

```diff
@@ -88,9 +107,15 @@ class DetectorCoordinateMap
 	stir::CartesianCoordinate3D<float> get_coordinate_for_det_pos( const stir::DetectionPosition<>& det_pos ) const
 	{ 
 		auto coord = det_pos_to_coord.at(det_pos);
-		coord.x() += static_cast<float>(distribution(generator));
-		coord.y() += static_cast<float>(distribution(generator));
-		coord.z() += static_cast<float>(distribution(generator));
+#ifdef STIR_OPENMP
+    auto thread_id = omp_get_thread_num();
+#else
+    auto thread_id = 0;
+#endif
+    
+		coord.x() += static_cast<float>(distributions[thread_id](generator));
+		coord.y() += static_cast<float>(distributions[thread_id](generator));
+		coord.z() += static_cast<float>(distributions[thread_id](generator));
```

Comment:
```suggestion
                auto& distribution = distributions[thread_id];
		coord.x() += static_cast<float>(distribution(generator));
		coord.y() += static_cast<float>(distribution(generator));
		coord.z() += static_cast<float>(distribution(generator));
```

However, this raises the question if `generator` is thread-safe. I have no idea, but somehow I think it's that one that isn't, while the distribution itself is. Not sure!

Reply:
good that you asked! I initially thought it was just the distribution (and the intel inspector complaints have gone away), but some more research showed that the generator is also not thread-safe. I'll just store a vector of instances for each.